openmdao/visualization/tables/table_builder.py

```diff
@@ -0,0 +1,1542 @@
+"""
+Table building classes.
+"""
+
+import sys
+import os
+import json
+import textwrap
+from itertools import zip_longest, chain
+from html import escape
+from io import StringIO
+from math import floor
+from numbers import Number, Integral
+from openmdao.utils.notebook_utils import notebook, display, HTML, IFrame, colab
+from openmdao.utils.file_utils import text2html
+from openmdao.utils.om_warnings import issue_warning
+
+
+_align2symbol = {
+    'center': '^',
+    'right': '>',
+    'left': '<'
+}
+
+_default_align = {
+    'int': 'right',
+    'real': 'right',
+    'bool': 'center',
+    'other': 'left',
+}
+
+_big_table_height = 600
+_row_height_px = 45  # use to size the IFrame in a notebook
+
+
+def _num_cols(rows):
+    ncols = None
+    for row in rows:
+        i = 0
+        for _ in row:
+            i += 1
+        if ncols is None or ncols < i:
+            ncols = i
+    return 0 if ncols is None else ncols
+
+
+class TableBuilder(object):
+    """
+    Base class for all table builders.
+
+    Parameters
+    ----------
+    rows : iter of iters
+        Data used to fill table cells.
+    headers : iter of str or None
```

Comment:
The code says that a value of 'keys' for headers does something special. Should the docs mention that ?

Reply:
fixed